src/overlays/actors/ovl_En_Elfgrp/z_en_elfgrp.c

```diff
@@ -181,236 +225,249 @@ void EnElfgrp_Init(Actor* thisx, PlayState* play) {
-
-    s32 temp_v1;
-
-    if ((arg1 < 0) || (arg1 >= 5)) {
+// Number of Stray Fairies in currently in Fountain
+s32 EnElfgrp_GetFountainFairiesCount(PlayState* play, s32 type) {
+    s32 numberInFountain;
+
+    if ((type < ENELFGRP_TYPE_MAGIC) || (type > ENELFGRP_TYPE_KINDNESS)) {
         return 0;
     }
 
-    if (arg1 == 0) {
+    if (type == ENELFGRP_TYPE_MAGIC) {
         if (gSaveContext.save.saveInfo.permanentSceneFlags[play->sceneId].unk_14 & 1) {
-            return 25;
+            return STRAY_FAIRY_TOTAL;
+        } else {
+            return STRAY_FAIRY_TOTAL - 1;
         }
-        return 24;
     }
 
-    temp_v1 = (gSaveContext.save.saveInfo.permanentSceneFlags[play->sceneId].unk_14 >> (((arg1 - 1) * 5) + 1)) & 0x1F;
-    if (temp_v1 < 10) {
-        temp_v1 = 10;
-    } else if (temp_v1 > 25) {
-        temp_v1 = 25;
+    numberInFountain =
+        (gSaveContext.save.saveInfo.permanentSceneFlags[play->sceneId].unk_14 >> (((type - 1) * 5) + 1)) & 0x1F;
+    if (numberInFountain < STRAY_FAIRY_TOTAL - STRAY_FAIRY_SCATTERED_TOTAL) {
+        numberInFountain = STRAY_FAIRY_TOTAL - STRAY_FAIRY_SCATTERED_TOTAL;
+    } else if (numberInFountain > STRAY_FAIRY_TOTAL) {
+        numberInFountain = STRAY_FAIRY_TOTAL;
     }
-    return temp_v1;
+    return numberInFountain;
 }
 
-void func_80A39CD4(PlayState* play, s32 arg1, s32 arg2) {
-    if ((arg1 < 0) || (arg1 > 4) || (arg2 < 10) || (arg2 > 25)) {
+// Update number of Stray Fairies in Fountain?
+void EnElfgrp_SetFountainFairiesCount(PlayState* play, s32 type, s32 newCount) {
+    if ((type < ENELFGRP_TYPE_MAGIC) || (type > ENELFGRP_TYPE_KINDNESS) ||
+        (newCount < (STRAY_FAIRY_TOTAL - STRAY_FAIRY_SCATTERED_TOTAL)) || (newCount > STRAY_FAIRY_TOTAL)) {
         return;
     }
 
-    if (arg1 == 0) {
-        if (arg2 == 25) {
+    if (type == ENELFGRP_TYPE_MAGIC) {
+        if (newCount == STRAY_FAIRY_TOTAL) {
             gSaveContext.save.saveInfo.permanentSceneFlags[play->sceneId].unk_14 |= 1;
         } else {
             gSaveContext.save.saveInfo.permanentSceneFlags[play->sceneId].unk_14 &= ~1;
         }
     } else {
-        gSaveContext.save.saveInfo.permanentSceneFlags[play->sceneId].unk_14 &= ~(0x1F << ((arg1 * 5) - 4));
-        gSaveContext.save.saveInfo.permanentSceneFlags[play->sceneId].unk_14 |= arg2 << ((arg1 * 5) - 4);
+        gSaveContext.save.saveInfo.permanentSceneFlags[play->sceneId].unk_14 &= ~(0x1F << ((type * 5) - 4));
+        gSaveContext.save.saveInfo.permanentSceneFlags[play->sceneId].unk_14 |= newCount << ((type * 5) - 4);
     }
 }
 
-void func_80A39DC8(EnElfgrp* this, PlayState* play, s32 arg2, s32 arg3) {
+void EnElfgrp_SpawnStrayFairies(EnElfgrp* this, PlayState* play, s32 count, s32 fairyType) {
     s32 pad;
     s32 i;
-    Actor* elforg;
-    s32 params;
-    Vec3f sp6C;
+    Actor* strayFairy;
+    s32 strayFairyParams;
+    Vec3f spawnCentrePos;
```

Comment:
Americanization suggestion:
```suggestion
    Vec3f spawnCenterPos;
```